### DISCUSSION TITLE
Fixing accessExternalDTD-Error in pom.xml

### DISCUSSION THREAD

**Discussion by CLandsberg** [quarkiverse/quarkus-cxf#1515] [Q&A]
Hallo,
i want to use wsdl2java which include dtd's and xsds. When i build the project it fails with 
`'file' access is not allowed due to restriction set by the accessExternalDTD property`

This can be easily fixed with adding a argument to the cli:
`mvn clean compile -Djavax.xml.accessExternalDTD=file`

**The question is how to add this to the pom.xml**? 
 It is not described in the Quarkus CXF User Guide.

I tried several ways like:

```
<plugin>
    <groupId>${quarkus.platform.group-id}</groupId>
    <artifactId>quarkus-maven-plugin</artifactId>
    <version>${quarkus.platform.version}</version>
    <extensions>true</extensions>
...
    <configuration>
        <jvmArgs>-Djavax.xml.accessExternalDTD=file</jvmArgs>
        <compilerArgs>-Djavax.xml.accessExternalDTD=file</compilerArgs>
    </configuration>
</plugin>
```

or the `quarkus.cxf.codegen.wsdl2java.additional-params` in the application.yml

But it did not work.

**Comment by ppalaga** [comment]
Thanks for the report, @CLandsberg. Quarkus CXF calls `wsdl2java` at build time, within the Maven process. Hence there is no way to set the sysprop in pom.xml or quarkus-maven-plugin config. 

The only way that could work is adding `-Djavax.xml.accessExternalDTD=file` to `.mvn/jvm.config` https://maven.apache.org/configure.html#mvn-jvm-config-file Maven passes those args when spawning a new Java process. 
Note that you enable it for the whole JVM which may bring substantial security concerns.

Anyway, a minimal reproducer would be nice, so that we are able to investigate what we could do at Quarkus CXF level.

**Comment by CLandsberg** [comment]
Thank you for the fast Answer.

Example WSDLs can be found under https://github.com/gematik/api-telematik/releases/tag/hsk_2.0.0
This together with `quarkus.cxf.codegen.wsdl2java.includes = *.wsdl`

**Reply by ppalaga** [reply]
There are few tens of WSDLs there. Would you be so kind and put together a minimal reproducer project that we could transform to an integration test once the issue is fixed?
